src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Azure.WebJobs.Script.WebHost.WebHooks;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public static class WebHostResolver
+    {
+        private static object _syncLock = new object();
+        private static ScriptHostConfiguration _scriptHostConfig;
+        private static WebScriptHostManager _hostManager;
+        private static SecretManager _secretManager;
+        private static WebHookReceiverManager _hookManager;
+        private static bool _prevStandbyMode;
+
+        public static ScriptHostConfiguration GetScriptHostConfiguration(WebHostSettings settings)
+        {
+            lock (_syncLock)
```

Comment:
Can't we have a fast path that doesn't require any locks? as in if current standby mode is false, then just return the object right away?